### ISSUE TITLE
[ENH] Implementing DBSCAN

### ISSUE THREAD

**Issue by aryanpola** [aeon-toolkit/aeon#2477]
### Describe the feature or idea you want to propose

We currently do not have a sub-module for density-based clustering. This issue is for implementing Density-Based Spatial Clustering of Applications with Noise (DBSCAN).

### Describe your proposed solution

Paper Reference: https://cdn.aaai.org/KDD/1996/KDD96-037.pdf

Scikit-learn for reference: https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/cluster/_dbscan.py

### Additional context

The `scikit-learn` implementation cannot `predict`. This does not fit with our interface.

**Comment by aryanpola** [comment]
@aeon-actions-bot assign @aryanpola

**Comment by SebastianSchmidl** [comment]
DBSCAN is already implemented in sklearn, and aeon depends on sklearn. Why do we need to re-implement it directly in aeon?

**Comment by aryanpola** [comment]
The clustering examples had a graph contained all the clustering sub-modules which includes density based, so figured why not implement it in aeon.
I'll close this if it's unnecessary.
Thanks for checking!

**Comment by MatthewMiddlehurst** [comment]
Agreed that unless there is some reason the sklearn version wont work (or is insufficient in any other way) that theres no need to have it here really. Any thoughts @chrisholder ?

**Comment by TonyBagnall** [comment]
the problem with dbscan is that it does not have a predict, i.e. it can only cluster train data. I'm not sure of the best way to deal with this. I'll have a go with the scikit learn version

**Comment by aryanpola** [comment]
> the problem with dbscan is that it does not have a predict, i.e. it can only cluster train data. I'm not sure of the best way to deal with this. I'll have a go with the scikit learn version

Yep, not having a predict function makes it a lot more difficult to cluster the newer points into the existing clusters (requires reevaluation of the model).
I'll close the issue then.

**Comment by TonyBagnall** [comment]
we cannot use DBSCAN in scikit directly, as it requires 2D numpy. We could wrap for unvariate and squash, but it would not work with multivariate. once #3613 is in we can port this in
